nabla/test/backstop.ts

```diff
@@ -304,85 +264,75 @@ export default async function (environment: TestSuiteEnvironment) {
       await backstop.withdraw(lpTokens + 1n, unit(19));
     },
 
-    async testNejc() {
-      const backstop2 = await newTestableBackstopPool(
-        address(router),
-        address(usd),
-        address(nablaCurve),
-        "Test Backstop LP",
-        "BLP"
-      );
-
-      await usd.approve(address(backstop2), MAX_UINT256);
-
-      //await backstop2.addSwapPool(address(swapPool1), 0);
-      //await backstop2.addSwapPool(address(swapPool2), 0);
-
-      vm.startPrank(BOB);
-      await usd.approve(address(backstop2), MAX_UINT256);
-      await usd.mint(BOB, MINT_AMOUNT);
-      await backstop2.deposit(MINT_AMOUNT);
-      vm.stopPrank();
-
-      const amount = e(1, 12);
-      const result = await backstop2.deposit(amount);
-      console.log("result", result);
-
-      const result2 = await backstop2.withdraw(amount, (amount * 99n) / 100n);
-      console.log("result2", result2);
-    },
-
-    // this test is also failing in the original test suite, so we exclude it here
-    async test_exclude_testBackstopLPWithdrawalInSwapLiquidity() {
+    async testBackstopLPWithdrawalInSwapLiquidity() {
       await depositInto(backstop, unit(20));
       const balanceBefore = await asset1.balanceOf(tester);
       const lpTokens = await backstop.balanceOf(tester);
 
-      const [backstopReservesBefore, backstopLiabBefore] = await backstop.coverage();
-      const [, swapLiabBefore] = await swapPool1.coverage();
+      const [reserve1, liability1] = await swapPool1.coverage();
+      const [reserve2, liability2] = await swapPool2.coverage();
+      const poolWorth =
+        (await usd.balanceOf(address(backstop))) +
+        (((reserve1 as bigint) - liability1) * PRICE_ASSET1) / PRICE_USD +
+        (((reserve2 as bigint) - liability2) * PRICE_ASSET2) / PRICE_USD;
+
+      assertEq(
+        poolWorth,
+        await backstop.getTotalPoolWorth(),
+        "Expect that backstop pool worth is computed as defined in spec"
+      );
+
+      const sharesToWithdraw = lpTokens / 2n;
+      const expectedReserveDecrement =
+        (((poolWorth * PRICE_USD) / PRICE_ASSET1) * sharesToWithdraw) / (await backstop.totalSupply());
+
+      const backstopReservesBefore = await usd.balanceOf(address(backstop));
+      const backstopLiabBefore = await backstop.totalSupply();
 
       // Check WithdrawSwapLiquidity event
-      vm.expectEmit(backstop, "WithdrawSwapLiquidity", [
-        tester,
-        address(swapPool1),
-        unit("2.000000000028515715"),
-        unit("10.000000000142578577"),
-      ]);
+      vm.expectEmit(backstop, "WithdrawSwapLiquidity", [tester, address(swapPool1), 2007413146409n, sharesToWithdraw]);
```

Comment:
The magic number here differs to the one used [here](https://github.com/0xamberhq/contracts/blob/8234ecc52bb2965625ee61cf78ba9d6d411a09c4/contracts/test/BackstopPool.t.sol#L376). How does the rounding work? Shouldn't it be `2007413146408n` if the number is truncated?

Reply:
The Nabla tests for Pendulum use 12 decimal assets where the Nabla tests for Ethereum use 18 decimal assets. I did it that way to ensure that Nabla also works correctly with 12 decimals (there were a few bugs before where this was not considered).

There is a slight numerical error between the numbers here due to the different number of decimals, that's why Nabla calculates 2007413146409n here instead of 2007413146408n.